app/services/search_service.rb

```diff
@@ -0,0 +1,28 @@
+class SearchService < ApplicationService
+  PROVIDERS = [Search::PbySearchProvider]
+
+  def call(query)
+    PROVIDERS.each do|klass|
+      provider = klass.new
+      items = provider.call(query.text)
+      source = provider.source
+
+      Query.transaction do
+        query.response_items.where(source: source).delete_all
```

Comment:
do I understand correctly that here you remove any previous results for this query from this source?
If so, would this not lose us the information about previously rejected results and previous favorited results? Or are they stored somewhere else?

Reply:
I'm not implemented rejecting results yet
 Actually there is something I'd like to discuss first.
If I understand properly rejected results should be shared across different queries? If so we'll need separate table for them.and we'll utilize external is/source columns for this

Reply:
The problem with adding this new table is that it probably should belongs to new project table we have discussed yesterday. And it should be added first as well.